pkg/controller/certificates/requestmanager/requestmanager_controller.go

```diff
@@ -394,7 +397,20 @@ func (c *controller) createNewCertificateRequest(ctx context.Context, crt *cmapi
 
 	if utilfeature.DefaultFeatureGate.Enabled(feature.StableCertificateRequestName) {
 		cr.ObjectMeta.GenerateName = ""
-		cr.ObjectMeta.Name = apiutil.DNSSafeShortenTo52Characters(crt.Name) + "-" + fmt.Sprintf("%d", nextRevision)
```

Comment:
Why did we shorten to 52 characters in the first place? Wasn't it too restrictive since the limit for object names is a DNS subdomain which allows for up to 253 characters?

For example, with the following 64 characters name:

```diff
-super-mega-long-certificate-name-that-uses-sixty-four-characters
+super-mega-long-certificate-name-that-uses--1
                                             ^ "-" + nextRevision
```

But it sounds like this was way below 253 chars, so we could have gone with the full name:

```diff
-super-mega-long-certificate-name-that-uses-sixty-four-characters
+super-mega-long-certificate-name-that-uses-sixty-four-characters-1
```

I think I'm missing the point of shortening to 52 characters

Reply:
The DNSSafeShortenTo52Characters function was blindly copied from the expression used to set the GenerateName value.
I explain here: https://github.com/cert-manager/cert-manager/blob/eac230f93efeb40bd1ff5e3d59c79fd7de641602/pkg/controller/certificates/requestmanager/requestmanager_controller.go#L381C1-L384 why this function was used.

Also, the first example would not occur, because DNSSafeShortenTo52Characters would have removed the last "-".

This PR removes the 52 character limit (because it was a mistake), instead we now make sure that the name is <= 253 characters.

Reply:
Ah. After scrolling to the end of the PR, I found why that is: the `generateName` field can only be given 53 characters:

```console
$ k create -f- <<EOF
apiVersion: v1
kind: Secret
metadata:
  generateName: super-mega-long-certificate-name-that-uses-sixty-four-characters
stringData:
  foo: bar
EOF
```

Output:

```bash
secret/super-mega-long-certificate-name-that-uses-sixty-four-charhgvrh created
#      <--------------------------------------------------------><--->
#                                  53 chars                      random
```

That's weird... I wonder why this limitation exists in the first place 😅

Reply:
> This PR removes the 52 character limit (because it was a mistake), instead we now make sure that the name is <= 253 characters.

Ah, right, since we aren't using `generateName` since this is the code for `StableCertificateRequestName `.

Reply:
I think we could actually provide a 58 character generateName without being truncated (see https://github.com/cert-manager/cert-manager/blob/eac230f93efeb40bd1ff5e3d59c79fd7de641602/pkg/controller/certificates/requestmanager/requestmanager_controller.go#L381C1-L384 and https://github.com/kubernetes/apiserver/blob/696768606f546f71a1e90546613be37d1aa37f64/pkg/storage/names/generate.go).

I think we are using DNSSafeShortenTo52Characters because that is a function that already existed.
But DNSSafeShortenTo57Characters would also work.